src/DotNet/Emit/DynamicMethodBodyReader.cs

```diff
@@ -186,9 +187,10 @@ public DynamicMethodBodyReader(ModuleDef module, object obj, Importer importer,
 			if (code is null)
 				throw new Exception("No code");
 			codeSize = code.Length;
-			var delMethod = rslvMethodFieldInfo.Read(obj) as SR.MethodBase;
+			var delMethod = rslvMethodFieldInfo.Read(obj) as DynamicMethod;
```

Comment:
Did you test .NET Framework 3.5 and 4.x?

Reply:
I checked the source code of `mscorlib` using dnSpy for both 2.0.0.0 and 4.0.0.0 versions and the field is of type `DynamicMethod`.